cmd/erasure-multipart.go

```diff
@@ -1151,9 +1153,13 @@ func (er erasureObjects) CompleteMultipartUpload(ctx context.Context, bucket str
 	}
 
 	// Save successfully calculated md5sum.
-	fi.Metadata["etag"] = opts.UserDefined["etag"]
+	// for replica, newMultipartUpload would have already sent the replication ETag
 	if fi.Metadata["etag"] == "" {
-		fi.Metadata["etag"] = getCompleteMultipartMD5(parts)
+		if opts.UserDefined["etag"] != "" {
+			fi.Metadata["etag"] = opts.UserDefined["etag"]
+		} else { // fallback if not already calculated in handler.
+			fi.Metadata["etag"] = getCompleteMultipartMD5(parts)
+		}
 	}
```

Comment:
unable to understand what does this logically change?

Reply:
for replication, the fi.Metadata already has the replication ETag passed in NewMultipartUpload and should not be overridden if it is set. 
https://github.com/minio/minio/pull/15346 was setting "etag" in in opts.UserDefined at the handler and passing it down to object layer, clobbering what was set in fi.Metadata already for replication